source/docs/casper/resources/tutorials/advanced/cross-contract.md

```diff
@@ -0,0 +1,538 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Cross Contract Communication
+
+Level of the tutorial: `Advanced`
+
+Time to complete the tutorial: `45 - 60 Minutes`
+
+Important topics to read before going further:
+
+- [getting started](../../../developers/writing-onchain-code/getting-started.md)
+- [writing on-chain code](../../../developers/writing-onchain-code/simple-contract.md)
+- [calling contracts](../../../developers/writing-onchain-code/calling-contracts.md)
+
+This will assure that you have needed tools and base knowledge to complete this tutorial. If not everything was clear while reading these sections, don't worry as we will be doing a full walkthrough step by step.
+
+## Outline of the tutorial {#outline}
+
+This tutorial covers all possible variations of cross contract communication. While it not necessary when writing a simple contract which doesn't have to interact with other deployed contracts it is nontheless important to understand the ways to achieve this feat.
+
+On a very high level a smart contract behaves like a class in object oriented programming. This would suggest, that the principles for object oriented programming also apply for smart contract development. What interests us the most is the responsibility separation of classes. On a conceptual level for very simple functionalities there is no need to separate the logic between more smart contracts. It is advised to think about it and preplan the development. A smart contract on Casper Network can be always updated, but for more complex scenarios there is a need to fully understand how the project can be segrageted into smaller units of logical work.
```

Comment:
"it is advised to think about it and preplan ..." I imagine that's someting everyone should do in every language? I need you to tell me what you are trying to communicate here. It seems like a series of very general statements about software development which don't actually tell me anything about Casper Smart Contracts, apart form making the point that they behave like classes and therefore should be programmed like good OO classes. I think what you are trying to do is speak specifically to the concern about separation of responsibility, and make the point that for small and simple Caper projects, there is probably not a need to make the proiject more complex by splitting out logic into different contracts, but a developer should analyse what they are trying to achieve and make decisions accordingly.

The other point I would make here is that we haven't touched any code so far, I have not built a cross-contract call whatsoever,  and you are already giving me advice about the right and the wrong way to use this technique. This should be a "tip" or an addendum to the document, to read once we have actually mastered the technique, not right up top.

Reply:
Agreed as well, the suggestions for best practices should be at the end, after it is clear how to do the things we are trying to teach. Didn't include it at the end as it should be clear at that point what the best practice actually is.